custom_components/waste_collection_schedule/waste_collection_schedule/source/ville_quebec_qc_ca.py

```
@@ -0,0 +1,258 @@
+                    address_text = option.text.strip()
+                    option_value = option['value']
+                    formatted_address = f"{address_text}. Use unique_id: {option_value}"
+                    formatted_addresses.append(formatted_address)
+                    option_values[option_value] = address_text
+
+                # If unique_id is provided, use it to select the proper option
+                if self._unique_id and self._unique_id in option_values:
+                    # Submit form with the selected address
+                    form_data = {
+                        '__VIEWSTATE': result_soup.find('input', {'id': '__VIEWSTATE'})['value'],
+                        '__VIEWSTATEGENERATOR': result_soup.find('input', {'id': '__VIEWSTATEGENERATOR'})['value'],
+                        '__EVENTVALIDATION': result_soup.find('input', {'id': '__EVENTVALIDATION'})['value'],
+                        'ctl00$ctl00$contenu$texte_page$ucInfoCollecteRechercheAdresse$RechercheAdresse$ddChoix': self._unique_id,
+                        'ctl00$ctl00$contenu$texte_page$ucInfoCollecteRechercheAdresse$RechercheAdresse$btnChoix': 'Poursuivre'
+                    }
+                    search_response = session.post(URL, data=form_data)
+                    if search_response.status_code != 200:
+                        raise Exception(
+                            f"Failed to submit address selection, status code: {search_response.status_code}")
+
+                    result_soup = BeautifulSoup(
+                        search_response.text, 'html.parser')
+                else:
+                    # Raise the SourceArgAmbiguousWithSuggestions exception with formatted addresses
+                    raise SourceArgAmbiguousWithSuggestions(
+                        argument="street_number_and_name",
+                        value=self._street_number_and_name,
+                        suggestions=formatted_addresses,
+                    )
```

This isn't a good idea as suggestions will be added to the argument as dropdown list in the config GUI, You could raise a normal Exception/SourceArgException or a SourceArgRequiredWithSuggestions for the ID